### PR TITLE
Make the console quieter by default

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

Changes most/all the `console.log()` to use the proper channel depending on the message.

Makes the client a lot quieter for most users, while keeping debug messages and outputting errors as actual errors.

**Issues**

None

### CODE REVIEW

File: src/components/guide/guide.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ define(['require', 'inputManager', 'browser', 'globalize', 'connectionManager',
         var offset = newPct - left;
         var pctOfWidth = (offset / width) * 100;
 
-        //console.log(pctOfWidth);
+        //console.debug(pctOfWidth);
```
Comment (dkanada) [inline src/components/guide/guide.js]: remove or uncomment

File: src/components/htmlaudioplayer/plugin.js [hunk 2]
```diff
@@ -298,7 +298,7 @@ define(['events', 'browser', 'require', 'apphost', 'appSettings', 'htmlMediaHelp
 
             var errorCode = this.error ? (this.error.code || 0) : 0;
             var errorMessage = this.error ? (this.error.message || '') : '';
-            console.log('Media element error: ' + errorCode.toString() + ' ' + errorMessage);
+            console.error('Media element error: ' + errorCode.toString() + ' ' + errorMessage);
```
Comment (dkanada) [inline src/components/htmlaudioplayer/plugin.js]:
```suggestion
            console.error('media element error: ' + errorCode.toString() + ' ' + errorMessage);
```

File: src/components/htmlvideoplayer/plugin.js [hunk 3]
```diff
@@ -408,7 +408,7 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
             lrd.media.streamType = cast.receiver.media.StreamType.OTHER;
             lrd.media.customData = options;
 
-            console.log('loading media url into mediaManager');
+            console.debug('loading media url into mediaManager');
```
Comment (dkanada) [inline src/components/htmlvideoplayer/plugin.js]:
```suggestion
            console.debug('loading media url into media manager');
```

File: src/components/htmlvideoplayer/plugin.js [hunk 4]
```diff
@@ -460,11 +460,11 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
...
             }
 
-            console.log('loading playback url: ' + url);
-            console.log('contentType: ' + contentType);
+            console.debug('loading playback url: ' + url);
+            console.debug('contentType: ' + contentType);
 
             host.onError = function (errorCode) {
-                console.log("Fatal Error - " + errorCode);
+                console.error("Fatal Error - " + errorCode);
```
Comment (dkanada) [inline src/components/htmlvideoplayer/plugin.js]:
```suggestion
                console.error("fatal error: " + errorCode);
```

File: src/components/htmlvideoplayer/plugin.js [hunk 5]
```diff
@@ -911,7 +911,7 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
         function onError() {
             var errorCode = this.error ? (this.error.code || 0) : 0;
             var errorMessage = this.error ? (this.error.message || '') : '';
-            console.log('Media element error: ' + errorCode.toString() + ' ' + errorMessage);
+            console.error('Media element error: ' + errorCode.toString() + ' ' + errorMessage);
```
Comment (dkanada) [inline src/components/htmlvideoplayer/plugin.js]:
```suggestion
            console.error('media element error: ' + errorCode.toString() + ' ' + errorMessage);
```

File: src/components/htmlvideoplayer/plugin.js [hunk 6]
```diff
@@ -1327,7 +1328,7 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
 
         function setCurrentTrackElement(streamIndex) {
 
-            console.log('Setting new text track index to: ' + streamIndex);
+            console.debug('Setting new text track index to: ' + streamIndex);
```
Comment (dkanada) [inline src/components/htmlvideoplayer/plugin.js]:
```suggestion
            console.debug('setting text track index to: ' + streamIndex);
```

File: src/components/keyboardnavigation.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 define(["inputManager", "layoutManager"], function (inputManager, layoutManager) {
     "use strict";
 
-    console.log("keyboardnavigation");
+    console.debug("keyboardnavigation");
```
Comment (dkanada) [inline src/components/keyboardnavigation.js]: maybe remove this one

File: src/components/nowplayingbar/nowplayingbar.js [hunk 8]
```diff
@@ -582,7 +582,7 @@ define(['require', 'datetime', 'itemHelper', 'events', 'browser', 'imageLoader',
 
     function onPlaybackStart(e, state) {
 
-        //console.log('nowplaying event: ' + e.type);
+        //console.debug('nowplaying event: ' + e.type);
```
Comment (dkanada) [inline src/components/nowplayingbar/nowplayingbar.js]: remove or uncomment

File: src/libraries/apiclient/apiclientcore.js [hunk 9]
```diff
@@ -59,22 +59,37 @@ define(["events", "appStorage"], function(events, appStorage) {
...
+        if (!serverAddress) {
+            throw new Error("Must supply a serverAddress");
+        }
+
+        console.debug("ApiClient serverAddress: " + serverAddress);
+        console.debug("ApiClient appName: " + appName);
+        console.debug("ApiClient appVersion: " + appVersion);
+        console.debug("ApiClient deviceName: " + deviceName);
+        console.debug("ApiClient deviceId: " + deviceId);
+        this._serverInfo = {};
```
Comment (dkanada) [inline src/libraries/apiclient/apiclientcore.js]:
```suggestion

        this._serverInfo = {};
```

File: src/libraries/apiclient/localassetmanager.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ define(["filerepository", "itemrepository", "useractionrepository", "transferman
     "use strict";
 
     function getLocalItem(serverId, itemId) {
-        return console.log("[lcoalassetmanager] Begin getLocalItem"), itemrepository.get(serverId, itemId)
+        return console.debug("[localassetmanager] Begin getLocalItem"), itemrepository.get(serverId, itemId)
```
Comment (dkanada) [inline src/libraries/apiclient/localassetmanager.js]:
```suggestion
        return console.debug("localassetmanager: begin getLocalItem"), itemrepository.get(serverId, itemId)
```

File: src/scripts/site.js [hunk 11]
```diff
@@ -582,13 +582,13 @@ var AppInfo = {};
...
     function onAppReady(browser) {
-        console.log("Begin onAppReady");
+        console.debug("begin onAppReady");
 
         // ensure that appHost is loaded in this point
         require(['apphost', 'appRouter'], function (appHost, appRouter) {
             window.Emby = {};
 
-            console.log("onAppReady - loading dependencies");
+            console.debug("onAppReady - loading dependencies");
```
Comment (dkanada) [inline src/scripts/site.js]:
```suggestion
            console.debug("onAppReady: loading dependencies");
```

File: src/controllers/wizard/user.js [hunk 12]
```diff
@@ -10,7 +10,7 @@ define(["loading", "globalize", "dashboardcss", "emby-input", "emby-button", "em
     }
 
     function onUpdateUserComplete(result) {
-        console.log(result);
+        console.debug(result);
```
Comment (JustAMan) [inline src/controllers/wizard/user.js]:
```suggestion
        console.debug("user update complete: ", result);
```

File: src/libraries/apiclient/apiclientcore.js [hunk 13]
```diff
@@ -149,7 +164,11 @@ define(["events", "appStorage"], function(events, appStorage) {
 
     function setSocketOnClose(apiClient, socket) {
         socket.onclose = function() {
-            console.log("web socket closed"), apiClient._webSocket === socket && (console.log("nulling out web socket"), apiClient._webSocket = null), setTimeout(function() {
+            console.debug("web socket closed");
+            apiClient._webSocket === socket;
+            console.debug("nulling out web socket");
+            apiClient._webSocket = null;
```
Comment (JustAMan) [inline src/libraries/apiclient/apiclientcore.js]: this is a bug, should be
```js
if (apiClient._webSocket === socket) {
    console.debug("nulling out web socket");
    apiClient._webSocket = null;
}
```

File: src/libraries/apiclient/apiclientcore.js [hunk 14]
```diff
@@ -244,24 +263,24 @@ define(["events", "appStorage"], function(events, appStorage) {
...
             return instance.lastFetch = (new Date).getTime(), response.status < 400 ? "json" === request.dataType || "application/json" === request.headers.accept ? response.json() : "text" === request.dataType || 0 === (response.headers.get("Content-Type") || "").toLowerCase().indexOf("text/") ? response.text() : response : (onFetchFail(instance, request.url, response), Promise.reject(response))
         }, function(error) {
-            if (error ? console.log("Request failed to " + request.url + " " + (error.status || "") + " " + error.toString()) : console.log("Request timed out to " + request.url), error && error.status || !enableReconnection) throw console.log("Reporting request failure"), onFetchFail(instance, request.url, {}), error;
-            console.log("Attempting reconnection");
+            if (error ? console.error("Request failed to " + request.url + " " + (error.status || "") + " " + error.toString()) : console.error("Request timed out to " + request.url), error && error.status || !enableReconnection) throw console.error("Reporting request failure"), onFetchFail(instance, request.url, {}), error;
```
Comment (JustAMan) [inline src/libraries/apiclient/apiclientcore.js]: I wonder how many more of those uglified things are still in the codebase...

File: src/libraries/apiclient/appStorage.js [hunk 15]
```diff
@@ -2,7 +2,7 @@ define([], function() {
     "use strict";
 
     function onCachePutFail(e) {
-        console.log(e);
+        console.error(e);
```
Comment (JustAMan) [inline src/libraries/apiclient/appStorage.js]:
```suggestion
        console.error("cannot put to a cache: ", e);
```

File: src/libraries/apiclient/apiclientcore.js (hunk 14 above)
Comment (MrTimscampi) [reply src/libraries/apiclient/apiclientcore.js]: Apiclient seems to be the main offender. So hopefully, once we're able to take it out of the tree and use the JS Apiclient repo, there should be a lot less of these.

Still a bunch of ternary operators that need cleanup, as well as usage of the comma operator...

File: src/components/chromecast/chromecasthelpers.js [hunk 16]
```diff
@@ -132,7 +132,7 @@ define(['events'], function (events) {
                 var match;
 
                 while (match = linkRegExp.exec(text)) {
-                    // console.log(matches);
+                    // console.debug(matches);
```
Comment (dkanada) [inline src/components/chromecast/chromecasthelpers.js]: uncomment or remove

File: src/components/chromecast/chromecastplayer.js [hunk 17]
```diff
@@ -377,7 +377,7 @@ define(['appSettings', 'userSettings', 'playbackManager', 'connectionManager', '
     };
 
     CastPlayer.prototype.onPlayCommandSuccess = function () {
-        //console.log('Message was sent to receiver ok.');
+        //console.debug('Message was sent to receiver ok.');
```
Comment (dkanada) [inline src/components/chromecast/chromecastplayer.js]: definitely uncomment

File: src/components/chromecast/chromecastplayer.js [hunk 18]
```diff
@@ -386,7 +386,7 @@ define(['appSettings', 'userSettings', 'playbackManager', 'connectionManager', '
      */
     CastPlayer.prototype.onMediaDiscovered = function (how, mediaSession) {
 
-        //console.log("chromecast new media session ID:" + mediaSession.mediaSessionId + ' (' + how + ')');
+        //console.debug("chromecast new media session ID:" + mediaSession.mediaSessionId + ' (' + how + ')');
```
Comment (dkanada) [inline src/components/chromecast/chromecastplayer.js]: uncomment

File: src/components/chromecast/chromecastplayer.js [hunk 19]
```diff
@@ -417,7 +417,7 @@ define(['appSettings', 'userSettings', 'playbackManager', 'connectionManager', '
      */
     CastPlayer.prototype.setReceiverVolume = function (mute, vol) {
         if (!this.currentMediaSession) {
-            //console.log('this.currentMediaSession is null');
+            //console.debug('this.currentMediaSession is null');
```
Comment (dkanada) [inline src/components/chromecast/chromecastplayer.js]: uncomment

File: src/components/nowplayingbar/nowplayingbar.js [hunk 20]
```diff
@@ -627,7 +626,7 @@ define(['require', 'datetime', 'itemHelper', 'events', 'browser', 'imageLoader',
 
     function onPlaybackStopped(e, state) {
 
-        //console.log('nowplaying event: ' + e.type);
+        //console.debug('nowplaying event: ' + e.type);
```
Comment (dkanada) [inline src/components/nowplayingbar/nowplayingbar.js]: uncomment

File: src/components/nowplayingbar/nowplayingbar.js [hunk 21]
```diff
@@ -653,7 +652,7 @@ define(['require', 'datetime', 'itemHelper', 'events', 'browser', 'imageLoader',
 
     function onStateChanged(event, state) {
 
-        //console.log('nowplaying event: ' + e.type);
+        //console.debug('nowplaying event: ' + e.type);
```
Comment (dkanada) [inline src/components/nowplayingbar/nowplayingbar.js]: uncomment

File: src/libraries/apiclient/sync/mediasync.js [hunk 22]
```diff
@@ -2,9 +2,9 @@ define(["localassetmanager"], function(localassetmanager) {
     "use strict";
 
     function processDownloadStatus(apiClient, serverInfo, options) {
-        return console.log("[mediasync] Begin processDownloadStatus"), localassetmanager.resyncTransfers().then(function() {
+        return console.debug("mediasync: begin processDownloadStatus"), localassetmanager.resyncTransfers().then(function() {
             return localassetmanager.getServerItems(serverInfo.Id).then(function(items) {
-                console.log("[mediasync] Begin processDownloadStatus getServerItems completed");
+                console.debug("mediasync: begin processDownloadStatus getServerItems completed");
```
Comment (dkanada) [inline src/libraries/apiclient/sync/mediasync.js]: You really didn't need to do any of these, we need to remove the API client from source eventually anyway.

**Pull request merged**